hera_qm/ant_metrics.py

```diff
@@ -306,33 +344,40 @@ def load_antenna_metrics(metricsJSONFile):
 
 
 class Antenna_Metrics():
-    '''Object for holding relevant visibility data and metadata with interfaces to four
+    """Container for holding data and meta-data for ant metrics calculations.
+
+    Object for holding relevant visibility data and metadata with interfaces to four
     antenna metrics (two for identifying dead antennas, two for identifying cross-polarized ones),
     an iterative method for identifying one bad antenna at a time while keeping track of all
-    metrics, and for writing metrics to a JSON. Works on raw data from a single observation
-    with all four visibility polarizations.'''
+    metrics, and for writing metrics to a HDF5. Works on raw data from a single observation
+    with all four visibility polarizations.
+    """
 
     def __init__(self, dataFileList, reds, fileformat='miriad'):
-        '''Arguments:
+        """Initilize an Antenna_Metrics object.
+
+        Arguments:
         dataFileList -- List of data filenames of the four different visibility
                         polarizations for the same observation
         reds -- List of lists of tuples of antenna numbers that make up redundant baseline groups
         format -- default 'miriad'. Other options: 'uvfits', 'fhd', 'ms ' (see pyuvdata docs)
-        '''
 
-        self.data = UVData()
+        """
+
         if fileformat == 'miriad':
-            self.data.read_miriad(dataFileList)
+            self.hd = HERAData(dataFileList, filetype='miriad')
         elif fileformat == 'uvfits':
-            self.data.read_uvfits(dataFileList)
+            self.hd = HERAData(dataFileList, filetype='uvfits')
         elif fileformat == 'fhd':
-            self.data.read_fhd(dataFileList)
+            raise NotImplemented(str(fileformat) + 'not supported')
         else:
             raise ValueError('Unrecognized file format ' + str(fileformat))
-        self.ants = self.data.get_ants()
-        self.pols = [pol.lower() for pol in self.data.get_pols()]
-        self.antpols = [antpol.lower() for antpol in self.data.get_feedpols()]
-        self.bls = self.data.get_antpairs()
+
+        self.data, self.flags, self.nsamples = self.hd.read()
```

Comment:
I'm not too familiar with HERAData.read() either. Looking through hera_cal code it returns all the values seen there for each antenna stored in its list of ants as an orderedDict. so this could potentially return a lot of data.